ambry-cloud/src/main/java/com/github/ambry/cloud/AzureStorage.java

```diff
@@ -0,0 +1,194 @@
+/**
+ * Copyright 2023 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.cloud;
+
+import com.azure.storage.blob.BlobContainerClient;
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.cloud.azure.AzureCloudConfig;
+import com.github.ambry.cloud.azure.AzureMetrics;
+import com.github.ambry.clustermap.ClusterMap;
+import com.github.ambry.clustermap.ReplicaState;
+import com.github.ambry.config.CloudConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.replication.FindToken;
+import com.github.ambry.store.FindInfo;
+import com.github.ambry.store.MessageInfo;
+import com.github.ambry.store.MessageWriteSet;
+import com.github.ambry.store.Store;
+import com.github.ambry.store.StoreException;
+import com.github.ambry.store.StoreGetOptions;
+import com.github.ambry.store.StoreInfo;
+import com.github.ambry.store.StoreKey;
+import com.github.ambry.store.StoreStats;
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Implements methods to write {@link com.github.ambry.clustermap.AmbryPartition} in Azure blob storage.
+ */
+public class AzureStorage implements Store {
```

Comment:
For the class naming, isn't AzureBlobStore better than AzureStorage?
We are implementing the BlobStore with the Azure Storage. 

And AzureBlobStore matches with the v1 cloudBlobStore as well.

Reply:
I prefer that the names of all related classes begin with the same prefix, for eg AzureStorage in this case. V1 code is unreadable and readability for others is my goal in v2. It is easy to search for related classes using a common prefix with grep or IDE or any internal code browser. I can call it AzureStorageBlobStore if you want but I am not interested in following v1.

Reply:
Done. Renamed all classes with prefix AzureStore

Reply:
You probably misunderstood my meaning. I don't have concern with AzureXXXX.
For this class itself, "AzureStorage" means interface to provide Azure Storage APIs. But this class is specifically to implement BlobStore on Azure. So the AzureBlobStore naming is better. And the previous version uses CloudBlobStore which matches the interface as well.

Reply:
got it, understood.